src/emscripten-optimizer/simple_ast.h

```diff
@@ -838,15 +838,15 @@ struct JSPrinter {
   static char* numToString(double d, bool finalize=true) {
     if (std::isnan(d)) {
       if (std::signbit(d)) {
-        return (char*) "-NaN";
+        return (char*) "-nan";
```

Comment:
Unless I'm confused, I think this class is trying to print out valid JS, and `NaN` is already valid JS? This change seems to make it print valid wasm or C++ perhaps?

Reply:
I added a comment as well, but `NaN` and `Infinity` aren't valid asmjs as they need to be imported explicitly through the global environment, and the names `nan` and `infinity` here I think match what emscripten does today

Reply:
That'll cause test failures in the test suite currently because they assert that literal asm.js compilation succeeds, but should that assertion get removed?

Reply:
Yeah, good point, we need to update the test suite to not require asm.js validation.

Also good point that might be a good topic for a separate PR. If so, how about adding `FIXME` comments on the things that are changed here that that PR will need to undo, like the `NaN` change? Unless you think you can get to undoing those very quickly?

Reply:
Sure thing, added some TODO annotations!